onnxruntime/core/providers/nnapi/nnapi_builtin/builders/helper.cc

```diff
@@ -306,33 +320,62 @@ bool HasValidQuantizationZeroPoints(const InitializedTensorSet& initializers, co
   return true;
 }
 
-common::Status GetQuantizationScale(const InitializedTensorSet& initializers, const Node& node,
-                                    size_t idx, float& scale) {
-  std::vector<uint8_t> unpacked_tensor;
-  const auto& name = node.InputDefs()[idx]->Name();
-  const auto& scale_tensor = *initializers.at(name);
-  ORT_RETURN_IF_ERROR(
-      onnxruntime::utils::UnpackInitializerData(scale_tensor, node.ModelPath(), unpacked_tensor));
-
-  // The scale should be one or more floats
-  ORT_RETURN_IF(unpacked_tensor.size() < 4, "The initializer [", name, "] should have one or more floats ",
-                "with size no less than 4, actual size: ", unpacked_tensor.size());
-  scale = reinterpret_cast<const float*>(unpacked_tensor.data())[0];
+common::Status GetQuantizationScaleAndZeroPoint(
+    const InitializedTensorSet& initializers, const NodeUnitIODef& io_def, const Path& model_path,
+    float& scale, int32_t& zero_point) {
+  scale = 0.0f;
+  zero_point = 0;
+
+  if (!io_def.quant_param) {  // Not a quantized IO
+    return ORT_MAKE_STATUS(ONNXRUNTIME, INVALID_ARGUMENT,
+                           "NodeArg: ", io_def.node_arg.Name(), " is not quantized");
+  }
+
+  const auto unpack_tensor = [&model_path](const InitializedTensorSet& initializers,
+                                           const std::string& name, std::vector<uint8_t>& unpacked_tensor) {
+    unpacked_tensor.clear();
```

Comment:
>unpacked_tensor [](http://example.com/codeflow?start=4&length=15)

is it necessary to clear the vector here? it looks like utils::UnpackInitializerData() will first call vector::resize()

Reply:
Just an old habit of mine, to clear any outbound variables first in the function
Consider it's internal, it's not necessary